src/main/java/microsoft/exchange/webservices/data/property/complex/time/TimeZoneDefinition.java

```diff
@@ -114,13 +115,38 @@ public int compare(final TimeZoneTransition x, final TimeZoneTransition y) {
     if (x == y) {
       return 0;
     } else if (x != null && y != null) {
-      final AbsoluteDateTransition firstTransition = (AbsoluteDateTransition) x;
-      final AbsoluteDateTransition secondTransition = (AbsoluteDateTransition) y;
-
-      final Date firstDateTime = firstTransition.getDateTime();
-      final Date secondDateTime = secondTransition.getDateTime();
-
-      return firstDateTime.compareTo(secondDateTime);
+			if (x instanceof AbsoluteDateTransition && y instanceof AbsoluteDateTransition) {
```

Comment:
@serious6 I had a quick look at how to create unit tests and found small problem - all three used class (AbsoluteDateTransition, RelativeDayOfMonthTransition and AbsoluteDayOfMonthTransition) have either protected or default visibility. They can only be instantiated by using TimeZoneTransition.create(...) method, which only create default instance of classes without any values. Values can only be set by using tryReadElementFromXml(...).

My question is following - shall I change visibility of set methods (which does not make much sense by TimeZoneDefenition) or is where any other simple method to create object of required classes with values (mocking EwsServiceXmlReader)? Or is where a general guideline for such cases?